Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -79,6 +79,20 @@ public final class SwiftLanguageServer: ToolchainLanguageServer {
     response: SKResponseDictionary,
     for snapshot: DocumentSnapshot)
   {
+    // Disable diagnostics for virtual files. Some editors (like Visual Studio Code) use virtual
+    // files to manage diff bases for the active document, which can lead to duplicate
+    // diagnostics in the Problems view.
+    //
+    // Additionally, as of Xcode 11.4, sourcekitd does not properly handle virtual files when
```

Comment:
Have we got a bug report for this, because we should figure out a real solution for this part.

Reply:
Filed https://bugs.swift.org/browse/SR-12196